lib/rule-tester/rule-tester.js

```diff
@@ -50,6 +50,52 @@ const
 
 const ajv = require("../shared/ajv")({ strictDefaults: true });
 
+
+//------------------------------------------------------------------------------
+// Typedefs
+//------------------------------------------------------------------------------
+
+/**
+ * A test case that is expected to pass lint.
+ * @typedef {Object} ValidTestCase
+ * @property {string} code Code for the test case.
+ * @property {any[]} [options] Options for the test case.
+ * @property {{ [name: string]: any }} [settings] Settings for the test case.
+ * @property {string} [filename] The fake filename for the test case. Useful for rules that make assertion about filenames.
+ * @property {string} [parser] The absolute path for the parser.
+ * @property {{ [name: string]: any }} [parserOptions] Options for the parser.
+ * @property {{ [name: string]: "readonly" | "writable" | "off" }} [globals] The additional global variables.
+ * @property {{ [name: string]: boolean }} [env] Environments for the test case.
+ */
+
+/**
+ * A test case that is expected to fail lint.
+ * @typedef {Object} InvalidTestCase
+ * @property {string} code Code for the test case.
+ * @property {number | Array<TestCaseError | string>} errors Expected errors.
```

Comment:
I've never used that, but by [this line](https://github.com/eslint/eslint/blob/3fa39a633b37544fec7cedfc1f2b0e62e9312a72/lib/rule-tester/rule-tester.js#L523) it seems that this array can also accept regex elements.

Reply:
Yes it is [documented](https://eslint.org/docs/developer-guide/nodejs-api#ruletester).
```suggestion
 * @property {number | Array<TestCaseError | string | RegExp>} errors Expected errors.
```